bionic/protocols.py

```diff
@@ -197,6 +198,39 @@ def __repr__(self):
         return f"{self.__class__.__name__}(...)"
 
 
+class JsonProtocol(BaseProtocol):
+    """
+    Decorator indicating that an entity's values can be serialized to the
```

Comment:
I just realized that it might be good to specify the exact set of types we support here, since "can be serialized to JSON" is a bit vague.